static/templates/stream_settings/change_general_stream_settings.hbs

```diff
@@ -33,18 +33,79 @@
+        <div class="input-group stream-privacy-values">
+            <div class="alert stream-privacy-status"></div>
+            <h4>{{t 'Who can access the stream?'}}
+                {{> ../help_link_widget link="/help/stream-permissions" }}
+            </h4>
+            {{#each ../stream_privacy_policy_values}}
+                <div class="radio-input-parent">
+                    <label class="radio">
+                        <input type="radio" name="privacy" value="{{ this.code }}" {{#if (eq this.code ../../stream_privacy_policy) }}checked{{/if}} />
+                        <b>{{ this.name }}:</b> {{ this.description }}
+                    </label>
+                </div>
+            {{/each}}
+        </div>
+
+        <div class="input-group">
+            <h4>{{t 'Who can post to the stream?'}}
+                {{> ../help_link_widget link="/help/stream-sending-policy" }}
+            </h4>
+            {{#each ../stream_post_policy_values}}
+                <div class="radio-input-parent">
+                    <label class="radio">
+                        <input type="radio" name="stream-post-policy" value="{{ this.code }}" {{#if (eq this.code ../stream_post_policy) }}checked{{/if}} />
+                        {{ this.description }}
+                    </label>
+                </div>
+            {{/each}}
+        </div>
+
+        <div>
+            <h4>{{t "Message retention period" }}
+                {{> ../help_link_widget link="/help/message-retention-policy" }}
+            </h4>
+
+            {{> ../settings/upgrade_tip_widget
+              zulip_plan_is_not_limited=../zulip_plan_is_not_limited}}
+
+            <div class="radio-input-parent">
+                <label class="radio">
+                    <input type="radio" name="message_retention" class="realm_default" value="realm_default" {{#if (eq "realm_default" ../stream_message_retention_days) }}checked{{/if}}
+                      {{#unless ../is_owner}}disabled{{/unless}}/>
+                    Use organization level settings {{../org_level_message_retention_setting}}
+                </label>
+            </div>
+            <div class="radio-input-parent">
+                <label class="radio">
+                    <input type="radio" name="message_retention" class="unlimited" value="unlimited" {{#if (eq "unlimited" ../stream_message_retention_days) }}checked{{/if}}
+                      {{#unless ../is_owner}}disabled{{/unless}}/>
+                    Retain forever
+                </label>
+            </div>
+            <div class="radio-input-parent">
+                <label class="radio">
+                    <input type="radio" name="message_retention" class="retain_for_period" value="retain_for_period" {{#if (eq "retain_for_period" ../stream_message_retention_days) }}checked{{/if}}
+                      {{#unless ../is_owner}}disabled{{/unless}}/>
+                    Retain for N days after posting
+                </label>
+            </div>
```

Comment:
Have we discussed to change this from dropdowns to radio buttons? If we have decided to do this, should we be changing the stream creation form too for consistency?

Reply:
No, i did not discuss this yet, but all the other settings have radio buttons, so i changed this too, so that it looks consistent.
Personally i prefer this, but i can also change it back to the dropdown. But i could also create another PR that changes the stream creation form